src/thread_pool_pppcsr/thread_pool_pppcsr.cpp

```diff
@@ -0,0 +1,110 @@
+//
+// Created by Christian Menges.
+//
+
+#include "thread_pool_pppcsr.h"
+#include <vector>
+#include <thread>
+#include <queue>
+#include <mutex>
+#include <iostream>
+#include <ctime>
+#include <numa.h>
+#include <cmath>
+
+using namespace std;
+
+/**
+ * Initializes a pool of threads. Every thread has its own task queue.
+ */
+ThreadPoolPPPCSR::ThreadPoolPPPCSR(const int NUM_OF_THREADS, bool lock_search) {
+  tasks.resize(NUM_OF_THREADS);
+  pcsr = new PPPCSR(456627.0, 456627.0, lock_search);
```

Comment:
These values are hard-coded leftovers for the vertex count of higgs twitter dataset. Should be changed to a user-defined variable (or automatically extracted from the dataset) in order to work with any given dataset. Low priority.